### PR TITLE
Significant plugin.yml & misc. updates

### PR DESCRIPTION
**Pull request by Tsuser1**
This pull request includes significant changes to the plugin.yml being included. These changes were implemented for the benefit of those using modern permissions management plugins, such as LuckPerms, which include information derived from the plugin.yml permissions. Additionally, there were minor changes made to pom.xml to allow building without needing to manually include external libraries, and the renaming of one class to be in-line with Java best practices.

Full changelog:

1. Updated .gitignore to include commonly mis-committed files
2. Fixed plugin.yml and added permission description tags for modern permissions plugins to display. Also added the api-version.
3. Updated pom.xml to include missing libraries and to update the archaic Spigot 1.12.2 API to Spigot 1.18 (latest stable version).
4. Renamed onPlayerMoveEvent to OnPlayerMoveEvent

### CODE REVIEW

**Comment by Tsuser1** [comment]
**New commit added: Modernized README.md**

Made the readme look much nicer.

**Comment by Tsuser1** [comment]
**New commit added: Fixed outdated Material**

Fixed an outdated reference to `Materal.FIREWORK` by replacing with the equivalent `Material.FIREWORK_ROCKET`.

**Comment by Tsuser1** [comment]
**Many patches, bugfixes, and lots of code review**

Review commits to see individual small fixes. The only one of major note is 95d6eea where a potential NullPointerException is fixed.

**Comment by MartenM** [comment]
Hey!

Thanks for your contributions to the project. I will look into this as soon as possible.

**Comment by MartenM** [comment]
I plan on removing support for everything lower then 1.12.
Do you know how these changes affect other versions? I considered using XMaterial for this project but haven't gotten around to implement it yet.

File: src/main/java/nl/martenm/servertutorialplus/ServerTutorialPlus.java [hunk 1]
```diff
@@ -96,8 +114,9 @@ public void onEnable(){
         lockedViews = new ArrayList<>();
         blockPlayers = new ArrayList<>();
         clickManager = new ClickManager(this);
-
-        MetricsLite metricsLite = new MetricsLite(this);
+        
+        // unused and disabled -- may be removed soon!
+        //MetricsLite metricsLite = new MetricsLite(this);
```
Comment (MartenM) [inline src/main/java/nl/martenm/servertutorialplus/ServerTutorialPlus.java:119]: The metrics class is not unused and is reporting correctly to bstats.org
https://bstats.org/plugin/bukkit/ServerTutorialPlus/382

File: src/main/java/nl/martenm/servertutorialplus/objects/NPCInfo.java [hunk 2]
```diff
@@ -71,4 +71,8 @@ public String getServerTutorialID() {
     public Location getLocation() {
         return location;
     }
+
+	public ServerTutorialPlus getPlugin() {
```
Comment (MartenM) [inline src/main/java/nl/martenm/servertutorialplus/objects/NPCInfo.java:75]: What is the use of putting a getter for the plugin instance here?

File: src/main/java/nl/martenm/servertutorialplus/managers/clickactions/ClickManager.java [hunk 3]
```diff
@@ -47,4 +47,8 @@ public void handleClickAction(PlayerInteractEntityEvent event){
         if(!hasClickaction(event.getPlayer().getUniqueId())) return;
         clickActions.get(event.getPlayer().getUniqueId()).run(event);
     }
+
+	public ServerTutorialPlus getPlugin() {
```
Comment (MartenM) [inline src/main/java/nl/martenm/servertutorialplus/managers/clickactions/ClickManager.java:51]: What is the use of getPlugin() here? If people want a plugin instance they should get it using the plugin manager or from the main plugin class.

File: src/main/java/nl/martenm/servertutorialplus/helpers/MetricsLite.java [hunk 4]
```diff
@@ -25,7 +25,6 @@
  *
  * Check out https://bStats.org/ to learn more about bStats!
  */
-@SuppressWarnings({"unchecked", "Convert2Lambda"})
```
Comment (MartenM) [inline src/main/java/nl/martenm/servertutorialplus/helpers/MetricsLite.java:28]: Please don't touch this class. This was the original MetricsLite class from BStats.
I will update this my self in another PR to be more up-to-date.

File: src/main/java/nl/martenm/servertutorialplus/events/OnPlayerInteractEventV1_8.java [hunk 5]
```diff
@@ -41,7 +41,7 @@ public void onPlayerInteractEvent(PlayerInteractEvent event){
             }
             ServerTutorial serverTutorial = PluginUtils.getTutorial(plugin, ts.ServerTutorialId);
             if(serverTutorial == null){
-                event.getPlayer().sendMessage(Lang.ERROR_FAILED_FINDING_TUTORIAL_ADMIN.toString().replace("%id%", serverTutorial.getId()));
+                event.getPlayer().sendMessage(Lang.ERROR_FAILED_FINDING_TUTORIAL_ADMIN.toString().replace("%id%", "NULL_ID"));
```
Comment (MartenM) [inline src/main/java/nl/martenm/servertutorialplus/events/OnPlayerInteractEventV1_8.java:44]: "NULL_ID"  should be `ts.ServerTutorialId`. Good catch finding this NPE though.

File: .settings/.gitignore [hunk 6]
```diff
@@ -0,0 +1 @@
+/org.eclipse.jdt.core.prefs
```
Comment (MartenM) [inline .settings/.gitignore:1]: I did rather not have the .settings folder in the GitHub directory.

**Review by MartenM — CHANGES_REQUESTED** [review]
Overall pretty okay changes. However, I wonder what this does for the plugins compatibility.
I don't want to quite drop support for 1.12 since 15% of the plugins are running on 1.12 servers.

**Pull request closed without merging**